src/views/Setup.vue

```diff
@@ -1,20 +1,98 @@
 <template>
   <div class="container">
-    <Card class="setup">
-      <div>
-        <p>We are preparing Sheikah</p>
+    <div class="center">
+      <div class="header">
+        <div class="filling-icon">
+          <font-awesome-icon
+            v-if="
+              setupMessage === 'Downloading wallet' ||
+                setupMessage === 'wallet up to date'
+            "
+            class="icon"
+            icon="wallet"
+          />
+          <font-awesome-icon
+            v-if="setupMessage === 'Running wallet'"
+            class="icon"
+            icon="cogs"
+          />
+          <font-awesome-icon
+            v-if="setupMessage === 'loaded'"
+            class="icon"
+            icon="check"
+          />
+          <div class="banner" />
+        </div>
+        <div>
+          <p class="progress-title">We are preparing Sheikah</p>
+          <p class="progress-subtitle">
+            {{ format(percentage) }}
+            <DotsLoading
+              v-if="!synced"
+              color="#444258"
+              data-test="loading-spinner"
+              class="spinner"
+            />
+          </p>
+        </div>
       </div>
-    </Card>
+      <el-progress
+        :text-inside="true"
+        :stroke-width="20"
+        :percentage="percentage"
+      />
+    </div>
   </div>
 </template>
 
 <script>
-import Card from '@/components/card/Card.vue'
+import { mapState, mapMutations } from 'vuex'
+import DotsLoading from '@/components/DotsLoading.vue'
 
 export default {
   name: 'WalletNotFound',
-  components: { Card },
-  methods: {},
+  components: {
+    DotsLoading,
+  },
+  computed: {
+    ...mapState({
+      setupMessage: state => state.uiInteractions.setupMessage,
+      client: state => state.wallet.api.client,
+      progress: state => state.uiInteractions.setupProgress,
+    }),
+    percentage() {
+      if (this.setupMessage === 'Downloading wallet') {
+        return this.progress / 4
```

Comment:
Avoid show float numbers
```suggestion
        return Math.round(this.progress / 4)
```